var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -85,6 +88,14 @@ class Espresso(Package):
     conflicts('+elpa', when='@:5.4.0')
     conflicts('+hdf5', when='@:5.4.0')
 
+    # To build with QMCPACK's hdf5 interface we version 5.3.0
+    # and hdf5
+    conflicts('+qmcpackconv', when='@5.4:')
```

Comment:
Then you either want:
```python
conflicts('+qmcpackconv', when='@5.3.1:')
conflicts('+qmcpackconv', when='@:5.2.999')
```
or no `conflicts()` at all. Unfortunately there is currently no _not_ in the spec syntax.